smack-extensions/src/main/java/org/jivesoftware/smackx/avatar/UserAvatarManager.java

```diff
@@ -0,0 +1,318 @@
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, PubSubException.NotALeafNodeException {
+        MetadataInfo info = new MetadataInfo(id, url, bytes, type, 0, 0);
+        publishAvatarMetadata(itemId, info, null);
+    }
+
+    /**
+     * Publish HTTP avatar metadata with its size in pixels.
+     *
+     * @param itemId
+     * @param id
+     * @param url
+     * @param bytes
+     * @param type
+     * @param pixelsHeight
+     * @param pixelsWidth
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public void publishHTTPAvatarMetadataWithSize(String itemId, String id, URL url, long bytes, String type,
+                                                  int pixelsHeight, int pixelsWidth)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, PubSubException.NotALeafNodeException {
+        MetadataInfo info = new MetadataInfo(id, url, bytes, type, pixelsHeight, pixelsWidth);
+        publishAvatarMetadata(itemId, info, null);
+    }
+
+    /**
+     * Publish avatar metadata.
+     *
+     * @param itemId
+     * @param id
+     * @param bytes
+     * @param type
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public void publishAvatarMetadata(String itemId, String id, long bytes, String type)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, PubSubException.NotALeafNodeException {
+        MetadataInfo info = new MetadataInfo(id, null, bytes, type, 0, 0);
+        publishAvatarMetadata(itemId, info, null);
+    }
+
+    /**
+     * Publish avatar metadata with its size in pixels.
+     *
+     * @param itemId
+     * @param id
+     * @param bytes
+     * @param type
+     * @param pixelsHeight
+     * @param pixelsWidth
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public void publishAvatarMetadataWithSize(String itemId, String id, long bytes, String type, int pixelsHeight,
+                                              int pixelsWidth)
+            throws NoResponseException, XMPPErrorException, NotConnectedException, InterruptedException, PubSubException.NotALeafNodeException {
+        MetadataInfo info = new MetadataInfo(id, null, bytes, type, pixelsHeight, pixelsWidth);
+        publishAvatarMetadata(itemId, info, null);
+    }
+
+    /**
+     * Get last data of an item.
+     *
+     * @param itemId
+     * @throws NoResponseException
+     * @throws XMPPErrorException
+     * @throws NotConnectedException
+     * @throws InterruptedException
+     */
+    public void requestLastData(String itemId)
```

Comment:
I'm not quite sure, what this method does. Does it store the latest date in the node cache?

Reply:
Dunno, remove it?